.github/workflows/ci-workflow.yml

```diff
@@ -0,0 +1,72 @@
+name: NPM CI
+on:
+  push:
+    paths-ignore:
+      - "docs/**"
+  pull_request:
+    paths-ignore:
+      - "docs/**"
+
+jobs:
+  npm-ci-job:
+    name: "YARN & NPM Build Test"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, windows-latest]
+    timeout-minutes: 25
+    steps:
+      # This step uses GitHub's hello-world-javascript-action: https://github.com/actions/hello-world-javascript-action
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '10.16'     
+      - name: Restore cached NODE modules
+        uses: actions/cache@v1
+        with:
+          path: node_modules
+          key: v4-dependencies-${{ hashFiles('yarn.lock') }}
+          restore-keys: |
+            v4-dependencies-${{ hashFiles('yarn.lock') }}
+            v4-dependencies- 
+      - name: "YARN install"
+        run: yarn install --frozen-lockfile
+      - name: "NPM run build"
+        run: npm run build && npm run build:types
+      - name: "Code linting"
+        run: npm run lint
+      - name: "Tests - headless browser"
+        run: npm run test:browser
+      - name: "Tests - BrowserStack"
+        run: npm run test:bs
+        if: >
+          !contains(toJson(github) , '"fork": true')
+        env:
+          BROWSER_STACK_USERNAME: ${{ secrets.BROWSER_STACK_USERNAME }}
+          BROWSER_STACK_ACCESS_KEY: ${{ secrets.BROWSER_STACK_ACCESS_KEY }}
+
+  # Publish job will be run only when the changes are merged to Master branch
+  npm-deploy-job:
+    name: "NPM Publish from Master"
+    runs-on: ubuntu-latest
+    needs: npm-ci-job
+    if: github.ref == 'refs/heads/master' && github.event	== 'push' && github.repository == 'open-wc/open-wc'
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '10.16'  
+          registry-url: https://registry.npmjs.org
+      - name: Restore cached NODE modules
+        uses: actions/cache@v1
+        with:
+          path: node_modules
+          key: v4-dependencies-${{ hashFiles('yarn.lock') }}
+          restore-keys: |
+            v4-dependencies-${{ hashFiles('yarn.lock') }}
+            v4-dependencies- 
+      - name: "Publish packages"
+        run: "npx lerna publish --message 'chore: releasing new packages versions' --yes"
```

Comment:
this is likely to fail when pushing tags, may also fail when publishing to npm.

I just this week (after trying longer than I'd care to admit) succeeded at [running `lerna publish` from github actions](https://github.com/apollo-elements/apollo-elements/blob/6e1bb2ab51a60cebbaed2c8aca86b2e8b2ec9336/.github/workflows/main.yml#L9-L54)

Reply:
I haven't yet reached that part though. Why should it fail?

Reply:
When GitHub Actions runs your workflow, the environment doesn't automatically have permissions to push commits or tags to your repository, which is what `lerna publish` does before it publishes to NPM.

Therefore, you have to create a GitHub Personal Access Token, store it in the repository's secrets, and reference it in the build step.

Similarly, in order to publish to NPM, you need to log in to the npm registry. It's not ideal to do this in CI using a username and password, so it's preferable to create an auth token on npmjs.com and reference it from the repo's secrets.

Both of these need to be configured in an incredibly specific way or they will fail.

If the process fails, the worst case scenario is that lerna will push tags to the repo and commit changes to `packages/**/package.json` (version bumps) and `packages/**/CHANGELOG.md`. These changes can be frustrating to undo in the case of failures.

Reply:
jup deploying probably should be tested with a custom registry first 🙈 
these things are always scary to change 😱 

locally we could use https://verdaccio.org/ but not sure how to reproduce it then? can you run actions locally? or can you define a registry that points to your locally running machine? (just for testing?)

Reply:
you can test a deploy to some local verdaccio on pull-request actions
I wouldn't draw much confidence from that, due to the finickiness of lerna, yarn, and friends